src/go/rpk/pkg/cli/cluster/partitions/balancer_run.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newTriggerBalnacerCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "balance",
+		Short: "Triggers on-demand partition balancer",
+		Long: `Triggers on-demand partition balancer.
+
+This command allows you to trigger on-demand partition balancer.
+
+With Redpanda Community Edition, partition count on each broker
+easily becomes uneven, which leads to data skew. In order to
+distribute partitions across brokers, user can run this command
+to trigger on-demand partition balancer.
+
+With Redpanda Enterprise Edition, even though Continuous Data Balancing
+monitors broker and rack availability, as well as disk usage, to avoid
+topic hotspots, there're still some edge cases where it can't handle well,
+i.e. a node becoming unavailable for a prolonged time and joining the
+cluster back thereafter. In such cases, user needs to run this command
+to trigger partition balancer manually.
+
+After you run this command, monitor the balancer progress using:
+
+    rpk cluster partitions balancer-status
+
+To see more detailed movement status, monitor the progress using:
+
+    rpk cluster partitions move-status
+`,
+
```

Comment:
```suggestion
With Redpanda Community Edition, the partition count on each broker
can easily become uneven, which leads to data skewing. To
distribute partitions across brokers, you can run this command
to trigger on-demand partition balancing.

With Redpanda Enterprise Edition, Continuous Data Balancing
monitors broker and rack availability, as well as disk usage, to avoid
topic hotspots. However, there are edge cases where users should manually trigger partition balancing (such as a node becoming unavailable for a prolonged time and rejoining the
cluster thereafter). In such cases, you should run this command
to trigger partition balancing manually.

After you run this command, monitor the balancer progress using:

    rpk cluster partitions balancer-status

To see more detailed movement status, monitor the progress using:

    rpk cluster partitions move-status
`,

```